lib/template/templates.h

```diff
@@ -93,14 +94,16 @@ struct _LogTemplateOptions
 void log_template_set_escape(LogTemplate *self, gboolean enable);
 gboolean log_template_set_type_hint(LogTemplate *self, const gchar *hint, GError **error);
 gboolean log_template_compile(LogTemplate *self, const gchar *template, GError **error);
-void log_template_format(LogTemplate *self, LogMessage *lm, const LogTemplateOptions *opts, gint tz, gint32 seq_num,
+void log_template_format(LogTemplate *self, LogMessage *lm, const LogTemplateOptions *opts, const LogStamp *processed,
+                         gint tz, gint32 seq_num,
                          const gchar *context_id, GString *result);
-void log_template_append_format(LogTemplate *self, LogMessage *lm, const LogTemplateOptions *opts, gint tz,
```

Comment:
I think the number of arguments passed here has surpassed the limit where we should encapsulate these into a struct and pass that.

We even had the beginnings of that in the form of LogTemplateInvokeArgs, that is only used during template function expansion, but it would be much better, if we extended its use into all call sites.

This would mean that even if we had further similar arguments, it would become a much smaller patch.

Reply:
I have this ancient branch, where I started this transition:

https://github.com/bazsi/syslog-ng/tree/wip/template-eval-arguments-as-a-struct-instead-of-parameters

I can't remember why I got distracted, maybe it was just time or I ran into some problems that I deemed was too difficult to solve.

Reply:
I have pushed some WIP commits.
I am not sure what would be the best.
- If LogTemplateEvalArgs has the same properties as in your branch, we could reduce the parameters of the log_template_append_format_with_context() and log_template_format_with_context() functions, and they looks good I guess.
- But in this case other functions in templates.h will have the same number of arguments as before, and it is ugly that we have similar functions with totally different argument list
- In the refactor commit I move messages and num_messages variables out of LogTemplateEvalArgs. In this case we can reduce all of these functions' parameters, and I guess it would look nice. I have not do that yet, because in this case we should modify so many files, and before every function call we have to create a structure filled with the parameters.
I am not sure what should I do, so I did finish non of these options.

Reply:
We agreed that the solution of this note needs a dedicated pull request.